src/QMCWaveFunctions/TrialWaveFunction.cpp

```diff
@@ -335,6 +333,28 @@ TrialWaveFunction::ValueType TrialWaveFunction::full_ratio(ParticleSet& P, int i
   return r;
 }
 
+TrialWaveFunction::RealType TrialWaveFunction::calcRatio(ParticleSet& P, int iat, ComputeType ct)
+{
+  ValueType r(1.0);
+  std::vector<WaveFunctionComponent*>::iterator it(Z.begin());
+  std::vector<WaveFunctionComponent*>::iterator it_end(Z.end());
+  for (int ii = V_TIMER; it != it_end; ++it, ii += TIMER_SKIP)
+  {
+    myTimers[ii]->start();
+    if (ct == ComputeType::ALL || ((*it)->is_fermionic && ct == ComputeType::FERMIONIC) || (!(*it)->is_fermionic && ct == ComputeType::NONFERMIONIC))
```

Comment:
I think this is the absolute limit for how complex a conditional should need to be.

Reply:
Do you want me to change something or this is just a comment?

Reply:
Just a comment. Although I think if we end up needing more logic here we should consider a different approach to dispatching this call.

Reply:
And the ComputeType tag sets us up to do that